### PR TITLE
Fixes #1145 Export spatial structure containers with individual and expressions

### LINKED ISSUE #1145: Export spatial structure containers with individual and expressions
There needs to be a workflow to export some combination of expression profiles, individual parameters, and container structure.

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #1145

# Description
Started out just allowing a user to select additional expression profiles to export with a container. We had similar functionality with selecting an individual. The issue I ran into was that not all containers for the expression profile existed in the containers during export. Those containers are only created during simulation creation.

I changed this feature to export Individual and Expression Profile parameters in a separate Parameter Values building block to capture all the parameters that need to be exported. 

I side project that came up as a result of this - we could not initially edit distributed parameters in a Parameter Values building block, but Individual building blocks could contain them, so I had to allow all the `PathAndValue` style building blocks to edit distributed parameters. That's the main reason the PR is so large. There was plenty of refactoring from that.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

Showing that you cannot export two expressions with the same molecule
image

# Questions (if appropriate):
https://github.com/Open-Systems-Pharmacology/MoBi/issues/1269

### CODE REVIEW

**Comment by rwmcintosh** [comment]
In the interest of not adding more to this huge PR, I opened #1269

File: src/MoBi.Presentation/Presenter/SelectFolderAndIndividualAndExpressionFromProjectPresenter.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+         IObjectPathFactory objectPathFactory,
+         IIndividualExpressionAndFilePathDTOMapper mapper) : base(view)
+      {
+         _buildingBlockRepository = buildingBlockRepository;
+         _editTaskForContainer = editTaskForContainer;
+         _objectPathFactory = objectPathFactory;
+         _mapper = mapper;
+      }
+
+      public (string, IndividualBuildingBlock, IReadOnlyList<ExpressionProfileBuildingBlock>) GetPathAndIndividualForExport(IContainer container)
```
Comment (msevestre) [inline src/MoBi.Presentation/Presenter/SelectFolderAndIndividualAndExpressionFromProjectPresenter.cs:56]: Not found of this for public interfaces
Is it possible to name parameters? I think so
string path, IndividualBuildingblock individual etc..

**Review by msevestre — COMMENTED** [review]
Looks good to me. Big PR so it needs to be tested 
Just a few remarks

**Review by msevestre — APPROVED** [review]

File: src/MoBi.Presentation/Presenter/SelectFolderAndIndividualAndExpressionFromProjectPresenter.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Presenter/SelectFolderAndIndividualAndExpressionFromProjectPresenter.cs:56]: I'm going to name the tuple members for the PR. Is it enough, or rather have a class for this?

**Pull request merged**